### PR TITLE
limit list of SFT servers to 5 (configurable), independent of the amount of servers …

### PR DESCRIPTION
**Pull request by jschaul**
…in the SRV record

Can be configured with

```yaml
sft:
  sftListLength: 17 # an integer outside the range 1,100 will throw an error at startup time
```

### CODE REVIEW

File: services/brig/src/Brig/Options.hs [hunk 1]
```diff
@@ -563,14 +566,17 @@ defSftServiceName = "_sft"
...
     let defaultV = 3600
         bounded = toBoundedInteger n :: Maybe Int64
-     in pure $
-          Timeout $
-            fromIntegral @Int $
-              maybe defaultV fromIntegral bounded
+     in pure
+          $ Timeout
+          $ fromIntegral @Int
+          $ maybe defaultV fromIntegral bounded
```
Comment (akshaymankar) [inline services/brig/src/Brig/Options.hs:579]: I think you're not using the right version of ormolu.

File: services/brig/src/Brig/Calling/API.hs [hunk 2]
```diff
@@ -124,14 +126,30 @@ newConfig env mSftEnv limit = do
...
+      -- priority and weight, but we internally agreed to randomize the list of
+      -- available servers for poor man's "load balancing" purposes.
+      -- FUTUREWORK: be smarter about list orderding depending on how much capacity SFT servers have.
+      randomizedSftEntries <- liftIO $ mapM randomize sftSrvEntries
+
+      -- Currently (Sept 2020) the client initiating an SFT call will try all
+      -- servers in this list. Limit this list to a smaller subset in case many
+      -- SFT servers are advertised in a given environment.
+      let subsetLength = Calling.sftListLength actualSftEnv
+      return $ subsetSft subsetLength <$> randomizedSftEntries
```
Comment (akshaymankar) [inline services/brig/src/Brig/Calling/API.hs:143]: Since all of this is _pure-ish_, we can unit test this.

**Review by akshaymankar — CHANGES_REQUESTED** [review]
Looks good, two requests:
1. Please add tests, I think we don't really need to integration test this, but unit testing should be trivial.
1. Please upgrade ormolu (should work with `stack install ormolu` from root of the repo) and reformat.

File: services/brig/src/Brig/Options.hs (hunk 1 above)
Comment (jschaul) [reply services/brig/src/Brig/Options.hs]: Fixed.

File: services/brig/src/Brig/Calling/API.hs (hunk 2 above)
Comment (jschaul) [reply services/brig/src/Brig/Calling/API.hs:142]: Let's maybe discuss in a call how best to test this without using an integration test; as this happens outside the easily-unit-testable discoveryloop logic.

**Review by akshaymankar — APPROVED** [review]

**Pull request merged**